### PR TITLE
Updating python support versions

### LINKED ISSUE #5006: Update Python versions page
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/faqs/Core/install-python-compatibility

### What part(s) of the page would you like to see updated?

Compatibility matrix:

| dbt-core → <br/> Python ↓ | 1.0 | 1.1 | 1.2 | 1.3 | 1.4 | 1.5 | 1.6 | 1.7 | 1.8 |
|---------------------------:|-----|-----|-----|-----|-----|-----|-----|-----|-----|
| **3.8**                   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   |
| **3.9**                   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   |
| **3.10**                  | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   | ✓   |
| **3.11**                  | ✖️   | ✖️   | ✖️   | ✖️   | ✓   | ✓   | ✓   | ✓   | ✓   |
| **3.12**                  | ✖️   | ✖️   | ✖️   | ✖️   | ✖️   | ✖️   | ✖️   | ✓   | ✓   |

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Updating the python version support FAQ with the most recent info.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/faqs/Core/install-python-compatibility.md [hunk 1]
```diff
@@ -5,26 +5,12 @@ sidebar_label: 'Python version'
 id: install-python-compatibility
 ---
 
-<VersionBlock firstVersion="1.6">
+Adapter plugins and their dependencies are not always compatible with the latest version of Python. For example, dbt-snowflake v0.15 is not compatible with Python 3.12, but dbt-snowflake versions 0.20+ are.
```
Comment (matthewshaver) [inline website/docs/faqs/Core/install-python-compatibility.md]: @aranke Is there a more recent example we can provide here?

**Comment by runleonarun** [comment]
@matthewshaver Can you fold your feedback into[ this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/5012)?

File: website/docs/faqs/Core/install-python-compatibility.md [hunk 2]
```diff
@@ -5,26 +5,11 @@ sidebar_label: 'Python version'
 id: install-python-compatibility
 ---
 
-<VersionBlock firstVersion="1.6">
+import Pythonmatrix from '/snippets/_python-compatibility-matrix.md';
 
-The latest version of `dbt-core` is compatible with Python versions 3.8, 3.9, 3.10 and 3.11.
+Use this table to match dbt-core versions with their compatible Python versions. New dbt minor versions will add support for new Python3 minor versions as soon as all dependencies can support it. In addition, dbt minor versions will drop support for old Python3 minor versions right before they reach [end of life](https://endoflife.date/python).
```
Comment (nghi-ly) [inline website/docs/faqs/Core/install-python-compatibility.md]: suggs: for conciseness and slightly more translation friendly

```suggestion
Use this table to match dbt-core versions with their compatible Python versions. New [dbt minor versions](/docs/dbt-versions/core#minor-versions) will add support for new Python3 minor versions when all dependencies can support it. In addition, dbt minor versions will withdraw support for old Python3 minor versions before their [end of life](https://endoflife.date/python).
```

**Review by nghi-ly — APPROVED** [review]
looks great! left one non-blocking sugg

**Pull request merged**